lib/processor.js

```diff
@@ -9,179 +9,51 @@
 "use strict";
 
 var assign = require("object-assign");
+var remark = require("remark");
+
+var SUPPORTED_SYNTAXES = ["js", "javascript", "node", "jsx"];
 
 var blocks = [];
 
 /**
- * Parses Markdown text for fenced code blocks containing JS.
- * @param {string} text Markdown text.
- * @returns {Block[]} List of parsed code blocks.
+ * Performs a depth-first traversal of the Markdown AST.
+ * @param {ASTNode} node A Markdown AST node.
+ * @param {object} callbacks A map of node types to callbacks.
+ * @returns {void}
  */
-function parse(text) {
-    var modes = {
-        NORMAL: "NORMAL",
-        CODE: "CODE"
-    };
-
-    var result = [],
-        index = 0,
-        line = 0,
-        column = 0,
-        mode = modes.NORMAL,
-        next;
-
-    /**
-     * Attempts to match a regex and advances if successful.
-     * @param {RegExp} test Regular expression to try to match.
-     * @returns {string[]|boolean} Match result or false if no match.
-     */
-    function match(test) {
-        var captures = test.exec(text.slice(index));
-
-        if (captures) {
-            index += captures[0].length;
-        }
-
-        return captures || false;
-    }
-
-    /**
-     * Tries to match an opening code fence on the current line.
-     * @returns {object|boolean} Fence data or false if no match.
-     */
-    function matchOpeningCodeFence() {
-        var captures;
-
-        // Exit early if an opening fence is not expected.
-        if (mode !== modes.NORMAL) {
-            return false;
-        }
-
-        captures = match(/^( {0,3})(`{3,}|~{3,})\h*(?:js|javascript|node|jsx)\h*\r?\n/i);
-        if (captures) {
-            return {
-                indent: captures[1],
-                fence: captures[2]
-            };
-        }
+function traverse(node, callbacks) {
```

Comment:
I thought about shaving that yak just for fun, but it was late, and I didn't want to write all the tests I'd have to write for a new module :)